binder/watcher.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binder
+
+import (
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type watcher interface {
+	// Returns a channel that sends events whenever workload mounts are created or removed
+	watch() <-chan workloadEvent
+}
+
+type Operation int
+
+const (
+	Added Operation = iota
+	Removed
+)
+
+const CredentialsSubdir = "creds"
+const CredentialsExtension = ".json"
+const PollSleepTime = 100 * time.Millisecond
+
+type workloadEvent struct {
+	// What happended to the workload mount?
+	op  Operation
+	uid string
+}
+
+func NewWatcher(path string) watcher {
+	return &pollWatcher{path}
+}
+
+type pollWatcher struct {
+	path string
+}
+
+func (p *pollWatcher) watch() <-chan workloadEvent {
+	c := make(chan workloadEvent)
+	go p.poll(c)
+	return c
+}
+
+func (p *pollWatcher) poll(events chan<- workloadEvent) {
+	// The workloads we know about.
+	known := make(map[string]bool)
+	credPath := filepath.Join(p.path + CredentialsSubdir)
```

Comment:
Can u merge the changes from https://github.com/colabsaumoh/proto-udsuspver/commit/63f881f8634f581f14b25430914b6ed0ac4ce204
at least the change in `watcher.go`.
When testing in my rig i had to make changes here to make it work.
other than that all of this looks good to me.

Reply:
Fixed!